java/arcs/core/data/proto/manifest.proto

```diff
@@ -156,6 +156,7 @@ message ReferenceTypeProto {
 message TypeVariableProto {
   string name = 1;
   ConstraintInfo constraint = 2;
+  bool maxType = 3;
```

Comment:
I think we should rename this to something like `accessAllFields`.

The reason is that the feature that `with {*}` represents is being able to access all fields from the resolved type. Making the type system choose the max type follows from that, but is not primary. E.g. If it was only the type system choosing the max type then DFA should not behave differently - but it's not about that - it's about what the particle can access, so DFA needs to be impacted.

If you argue that we should be consistent with the TypeScript side, then I would say that this ship has sailed already. In the protocol buffer we use type variables exclusively to signify generics in connections, which is why we decided to not have canReadSubset and canWriteSuperset. Instead we just have one constraint that represents fields that the particle will read. 

Maybe @bgogul can comment and convince me otherwise.

Reply:
I agree `maxType` is not quite intuitive. Also, this new field should go into the `TypeConstraintsProto` field as this essentially part of the constraints associated with the type variable? A couple of suggestions:
  - Add this field to `TypeConstraintsProto`.
  - Name it as `unconstrained` or `no_constraints`? (Alternatively, we can name it as `has_constraints`, which should be set to `false` if `maxType` is `true` and vice versa.)
  - In the Kotlin data class for `TypeVariable`, add a property `noConstraints` or `unconstrained` that is set appropriately. (Ideally, we want Kotlin to have a `TypeConstraints` class, but let us defer it for later as we will need to make some change in google3 simultaneously for that.)

Reply:
I was resistant to the term "unconstrained" on first read, but after looking at the code, I see its appeal. Should be implemented now.